src/Gui/Splashscreen.cpp

```diff
@@ -148,76 +113,143 @@ class SplashObserver : public Base::ILogger
             // ignore activation of commands
             rx.setPattern(QLatin1String("^\\s*(\\+App::|Create|CmdC:|CmdG:|Act:)\\s*"));
             match = rx.match(msg);
-            if (match.hasMatch() && match.capturedStart() == 0)
+            if (match.hasMatch() && match.capturedStart() == 0) {
                 return;
+            }
         }
+        msg.replace(QLatin1String("\n"), QString());
+
+        /**
+         * Previously SplashObserver got involved with the formatting of a message.
+         * This has been removed. SplashObserver has no responsibility for formatting (SRP)
+         */
+        splash.showMessage(msg);
 
-        splash->showMessage(msg.replace(QLatin1String("\n"), QString()), alignment, textColor);
         QMutex mutex;
         QMutexLocker ml(&mutex);
         QWaitCondition().wait(&mutex, 50);
     }
 
 private:
-    QSplashScreen* splash;
-    int alignment;
-    QColor textColor;
+    QSplashScreen& splash;
 };
-} // namespace Gui
+}// namespace Gui
 
 // ------------------------------------------------------------------------------
 
 /**
  * Constructs a splash screen that will display the pixmap.
  */
-SplashScreen::SplashScreen(  const QPixmap & pixmap , Qt::WindowFlags f )
-    : QSplashScreen(pixmap, f)
+SplashScreen::SplashScreen(const QPixmap& pixmap, Qt::WindowFlags f) : QSplashScreen {pixmap, f}
 {
-    // write the messages to splasher
-    messages = new SplashObserver(this);
+    const std::map<std::string, std::string>& config = App::Application::Config();
+
+    /**
+     * Breaking up code by SRP highlighted conditions that were not taken into account
+     * Even so, it is dubious that the following has to happen here
+     * This is a dependency that could perhaps better be injected
+     */
+
+    std::string colStr = fromConfig(config, "SplashTextColor");
+    if (!colStr.empty()) {
+        auto color = colorFromString(colStr);
+        if (color.isValid()) {
+            textColor = color;
+        }
+    }
+
+    std::string alignStr = fromConfig(config, "SplashAlignment");
+    if (!alignStr.empty()) {
+        alignment = alignStrToInt(alignStr);
+    }
+
+    splashObserver = new SplashObserver(*this);
 }
 
 /** Destruction. */
 SplashScreen::~SplashScreen()
 {
-    delete messages;
+    delete splashObserver;
+}
+
+void SplashScreen::showMessage(const QString& message)
+{
+    QSplashScreen::showMessage(message, alignment, textColor);
+}
+
+std::string SplashScreen::fromConfig(const std::map<std::string, std::string>& config,
+                                     const std::string value)
+{
+    auto al = config.find(value);
+    return al == config.end() ? "" : al->second.c_str();
+}
+
+QColor SplashScreen::colorFromString(const std::string str)

```

Comment:
Shouldn't it be `const std::string& str`? From what I have read a while ago in C++11 you can also declare it `std::string str` but then in the implementation you should use `std::move(str)` to avoid to create a tmp. copy.